### PR TITLE
Add f5-rest-node-lib sample workers

### PR DESCRIPTION
**Pull request by dajiabao**

### CODE REVIEW

File: workers/collections/authorWorker.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  either express or implied. See the License for the specific
+  language governing permissions and limitations under the License.
+*/
+
+
+"use strict";
+
+class AuthorWorker {
+    constructor() {
+        this.WORKER_URI_PATH = "shared/authors"; //demo purpose, final collection URI convention may change
```
Comment (elv1s) [inline workers/collections/authorWorker.js]: please remove the comment from this line, and the next

File: workers/collections/authorWorker.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            netWorth: item.netWorth
+        };
+    }
+
+    onGet(item) {
+        return {
+            id: item.id,
+            name: item.name,
+            books: item.books,
+            netWorth: this.netWorthDefaultUpdate(item)
```
Comment (elv1s) [inline workers/collections/authorWorker.js]: minor - please move this function off of the prorotype and into a private function

File: workers/fileWorker.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    getExampleState() {
+        return {
+            filePath: "pathOfFileToCheck"
+        };
+    }
+}
+
+module.exports = FileWorker;
```
Comment (elv1s) [inline workers/fileWorker.js]: minor - newline at end of file

File: workers/greeting/greetingWorker.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            this.completeRestOperation(restOperation);
+        };
+
+        q.spread([helloCall, worldCall], handleResolvedCalls.bind(this)).catch(handleError);
+    } catch (err) {
+        handleError(err);
+    }
+};
+
+module.exports = GreetingWorker;
```
Comment (elv1s) [inline workers/greeting/greetingWorker.js]: minor - newline at end of file

File: workers/persistedWorker.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
+/*
```
Comment (elv1s) [inline workers/persistedWorker.js]: can you remove this persistedWorker from the pull request. We want to encourage the collection/singleton worker pattern for stateful workers. Thanks!

File: workers/singletons/meWorker.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+     * @memberof MeWorker
+     */
+    getDefaultState() {
+        return {
+            FirstName: 'Joe',
+            LastName: 'Alonso',
+            UserName: 'alonso',
+            Emails: [
+                'j.alonso@f5.com',
+                'alonso@f5.com'
```
Comment (elv1s) [inline workers/singletons/meWorker.js]: can you change the name please?

File: workers/singletons/meWorker.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    /**
+     * Return odata schema for document
+     *
+     * @returns {Object}
+     * @memberof MeWorker
+     */
+    getSchema() {
+        return require('odata/sample/models/person');
```
Comment (elv1s) [inline workers/singletons/meWorker.js]: inline the schema please

File: workers/singletons/youWorker.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     * Return default state of document
+     *
+     * @returns {Object|Promise} default state at startup
+     * @memberof YouWorker
+     */
+    getDefaultState() {
+        return new Promise((resolve, reject)=> {
+            resolve({
+                FirstName: 'Keith',
```
Comment (elv1s) [inline workers/singletons/youWorker.js]: change name

File: workers/singletons/youWorker.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    /**
+     * Return odata schema for document
+     *
+     * @returns {Object}
+     * @memberof YouWorker
+     */
+    getSchema() {
+        return require('odata/sample/models/person');
```
Comment (elv1s) [inline workers/singletons/youWorker.js]: inline schema

**Review by elv1s — CHANGES_REQUESTED** [review]

File: workers/fileWorker.js (hunk 3 above)
Comment (dajiabao) [reply workers/fileWorker.js]: newline added

File: workers/greeting/greetingWorker.js (hunk 4 above)
Comment (dajiabao) [reply workers/greeting/greetingWorker.js]: newline added

**Review by elv1s — APPROVED** [review]
ship it!

**Pull request merged**